solar_system_allwaves.rb

```diff
@@ -0,0 +1,184 @@
+                                        * Focus on building a single class
+                        Wave 2:     Uses multiple classes that are composed together
+                        Wave 3:     A UI is built that allows the user to add own planets
+
+REQUIREMENTS: Wave 1:      1.  Build a single class, SolarSystem:
+                              * SolarSystem must have the instance variable, @planets
+                                      YUP
+                              * An initialize method should be created such that:
+                                 -- An array of planet names are stored in @planets
+                                       YUP
+                          2.  Two additional methods:
+                                 * One that contributes additional planets to the list
+                                    (via a **parameter**, not user input)
+                                        YUP
+                                * A second that will RETURN (not print) the ontents of @planets
+                                   in an ordered list.
+                                        YUP
+                          3.  Create code for testing
+                                SORT OF
+                          4.  "Instead of Strings for planets, modify SolarSystem's initialize
+                                method to take a list of hashes where each planet is sent as a hash with
+                                at least 5 attributes."
+                                    YES
+
+                Wave 2:
+                         1. Create a Planet class that will represent a planet.
+                            A.  Give Planet an initialize method that will take several arguments and use them
+                                to set the class's instance variables
+
+                                YUP
+                            B.  Create a method that returns the Planets attributes in an easy-to-read fashion
+
+                                YUP
+
+                            C.  Create reader methods that give the user access to read the instance variables
+
+                                YUP
+
+                         2. Make SolarSystem take an array of Planets instead of hashes
+
+                                YUP
+
+                         3. When printing the planet list or planet details, it should call the corresponding method in Planet.
+
+                                YUP
+
+                Wave 3:
+                        1. Create an interface where the user can interact with the solar system and be able to select a planet and view information about it.
+
+                                YUP
+
+                        2. Allow your user to add their own planet.
+
+                                YUP
+=end
+
+class SfWorld
+  attr_accessor :name, :author, :first_book, :year, :lifeforms
+  def initialize (name, author, first_book, year, lifeforms)
```

Comment:
This is just a small comment--

when I open this code in Atom on my computer, Atom puts a little squiggly underline underneath this line giving me a warning. (It doesn't affect my code running or anything, just gives me a warning). If you see that squiggly underline on your computer too and it bothers you, it's simply because Atom thinks there shouldn't be a space between `def initialize` and the `(` parenthesis character, and it wants it to look like:
```ruby
def initialize(name, author, first_book, year, lifeforms)
```
instead of
```ruby
def initialize (name, author, first_book, year, lifeforms)
```

like i said, not a big deal though :P